### PR TITLE
[GCS]Limit the number of profile table

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
https://github.com/ray-project/ray/issues/10354

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/stats_handler_impl.cc [hunk 1]
```diff
@@ -34,7 +35,7 @@ void DefaultStatsHandler::HandleAddProfileData(const AddProfileDataRequest &requ
     GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   };
 
-  Status status = gcs_table_storage_->ProfileTable().Put(UniqueID::FromRandom(),
+  Status status = gcs_table_storage_->ProfileTable().Put(ids_[cursor_++ % ids_.size()],
```
Comment (rkooo567) [inline src/ray/gcs/gcs_server/stats_handler_impl.cc:38]: I just need some clarification answer here. When you store data in this way, and run `ray timeline`, does it work as expected?

File: src/ray/common/ray_config_def.h [hunk 2]
```diff
@@ -208,6 +208,9 @@ RAY_CONFIG(uint32_t, maximum_gcs_deletion_batch_size, 1000)
 /// Maximum number of items in one batch to scan from GCS storage.
 RAY_CONFIG(uint32_t, maximum_gcs_scan_batch_size, 1000)
 
+/// Maximum number of rows in GCS profile table.
+RAY_CONFIG(uint32_t, maximum_profile_table_rows_count, 1000)
```
Comment (rkooo567) [inline src/ray/common/ray_config_def.h:212]: 1000 might be too low given each of ray call might generate 4~5 profile data. When you discovered the issue, how many rows were at the profile table?

File: src/ray/gcs/gcs_server/stats_handler_impl.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/gcs/gcs_server/stats_handler_impl.cc:38]: Also, please comment why we are doing this.

Comment (ffbin) [reply src/ray/gcs/gcs_server/stats_handler_impl.cc:38]: > I just need some clarification answer here. When you store data in this way, and run `ray timeline`, does it work as expected?

image
The first time I used `ray timeline`, I'm not sure if it work as expected.

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**